src/smallmap_gui.cpp

```diff
@@ -585,11 +585,18 @@ uint32_t GetSmallMapOwnerPixels(TileIndex tile, TileType t, IncludeHeightmap inc
 		case MP_VOID:     return MKCOLOUR_XXXX(PC_BLACK);
 		case MP_INDUSTRY: return MKCOLOUR_XXXX(PC_DARK_GREY);
 		case MP_HOUSE:    return MKCOLOUR_XXXX(PC_DARK_RED);
-		default:          o = GetTileOwner(tile); break;
-		/* FIXME: For MP_ROAD there are multiple owners.
-		 * GetTileOwner returns the rail owner (level crossing) resp. the owner of ROADTYPE_ROAD (normal road),
-		 * even if there are no ROADTYPE_ROAD bits on the tile.
-		 */
+		case MP_ROAD:
+			if (GetRoadTypeRoad(tile) != INVALID_ROADTYPE) {
+				o = GetRoadOwner(tile, RTT_ROAD);
+			}
+			/* Must be tram. */
```

Comment:
Is there an `else` missing? `o` is reassigned unconditionally.

Reply:
Yeah, I had been reworking the code and apparently pushed some intermediate version.